src/Nethermind.Int256/UInt256.cs

```diff
@@ -520,4120 +266,1186 @@ private static bool AddVector256(in UInt256 a, in UInt256 b, out UInt256 res)
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
     private static void AddWithCarry(ulong x, ulong y, ref ulong carry, out ulong sum)
     {
-        ulong result = x + y + carry;
-        // both msb bits are 1 or one of them is 1 and we had carry from lower bits
-        carry = ((x & y) | ((x | y) & (~result))) >> 63;
-        sum = result;
+        ulong t = x + y;
+        ulong r = t + carry;
+        carry = (t < x ? 1UL : 0UL) + (r < t ? 1UL : 0UL);
+        sum = r;
     }
 
```

Comment:
Would this be faster? Or is it already only in registers?

```
private static ulong AddWithCarry(ulong x, ulong y, ulong carry, out ulong sum)
{
    ulong t = x + y;
    sum = t + carry;
    return ((t < x) ? 1UL : 0UL) + ((sum < t) ? 1UL : 0UL);
}
```

Reply:
`sum` is address exposed as an `out` variable so using local register since it gets read back

In theory wouldn't matter because the inlining should eliminate it but the Jit still does bit of unnecessary work